universal-application-tool-0.0.1/app/services/question/ReadOnlyQuestionServiceImpl.java

```diff
@@ -21,16 +24,26 @@ public ReadOnlyQuestionServiceImpl(ImmutableList<QuestionDefinition> questions)
     ImmutableMap.Builder<Path, QuestionDefinition> questionPathMap = ImmutableMap.builder();
     ImmutableMap.Builder<Path, ScalarType> scalarMap = ImmutableMap.builder();
     ImmutableMap.Builder<Path, QuestionDefinition> scalarParentsMap = ImmutableMap.builder();
+    for (ImmutableList<QuestionDefinition> qds :
+        questions.stream()
+            .collect(new GroupByKeyCollector<>(QuestionDefinition::getName))
+            .values()) {
+      Optional<QuestionDefinition> qdMaybe =
+          qds.stream().filter(qd -> qd.getLifecycleStage().equals(LifecycleStage.ACTIVE)).findAny();
```

Comment:
Thoughts (blocking for an answer): It seems like an invariant that every question has an ACTIVE version. This might be worth documenting somewhere.

Having said that, is it possible to have a question that only has a version in DRAFT? Should we include those in pathMap? Please document the answer in comment here.

Reply:
That is not an invariant, but there is a related invariant: any question which has ever been included in at least one active program has exactly one active version.

Therefore any question which could plausibly have an aaswer is ACTIVE.

Reply:
Ah makes sense, we only need the pathMap in applicant service, where all the programs relevant are ACTIVE, and so are all the questions contained. Would you document this behavior in ReadOnlyQuestionService.java or here?